### PR TITLE
AMP-9 healtcheck follow up

### PR DESCRIPTION
**Pull request by wajerrr**
- `healtcheck` endpoint test added 
- refactored `config` files
- added `utils/environment-detection.js`
 - small fixes 'n' tidying up

### CODE REVIEW

File: src/server/config/base-config.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import packagejson from '../../../package.json';
+import { isDev } from '../utils/environment-detection';
+
+export default {
+  name: packagejson.name,
+  version: packagejson.version,
+  httpPort: process.env.PORT || 8000,
+  host: isDev === 'development' ? 'localhost' : '0.0.0.0',
```
Comment (dosyara) [inline src/server/config/base-config.js]: `isDev === 'development'` is this right? should be just `isDev` I think

File: src/server/index.js [hunk 2]
```diff
@@ -14,11 +16,11 @@ const start = async () => {
   }
   /* eslint-disable-next-line no-console */
   console.info(
-    `Server running at: ${
-      server.info.uri
-    } in ${process.env.NODE_ENV.toUpperCase()} mode`
+    `Server running at: ${server.info.uri} in ${(
+      process.env.NODE_ENV || 'no NODE_ENV set!!!'
```
Comment (dosyara) [inline src/server/index.js]: why so emotional? :)
throw error maybe? or use `production` as a default?

File: src/server/handlers/amp-page-render.js [hunk 3]
```diff
@@ -1,29 +1,32 @@
 import getGraphqlData from '../get-graphql-data';
 import renderHtml from '../render-html';
+import { isProd, isStage } from '../utils/environment-detection';
+
+import economistConfig from '../config/economist';
+
+/* This method will be called now every time we hit this endpoint
+ Maybe we should have env variable to detect what product it is
```
Comment (dosyara) [inline src/server/handlers/amp-page-render.js]: Regarding this comment and getDomain function. 
I assumed, we will define everything on startup. Each "product" (economist/worldif etc) will have each own web server and same for each environment (dev/stage/prod)

Comment (wajerrr) [reply src/server/handlers/amp-page-render.js]: yes maybe / I hope so but for now we are detecting product by host therefore executing detection method an every  request...

File: src/server/index.js (hunk 2 above)
Comment (wajerrr) [reply src/server/index.js]: well, jut don't like people not setting they NODE_ENV :-) now we exit if there is no NODE_ENV set

File: src/server/config/base-config.js (hunk 1 above)
Comment (wajerrr) [reply src/server/config/base-config.js]: done

File: src/server/config/economist.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export default {
+  domain: 'https://www.economist.com',
+  ampHost: 'amp.economist.com',
+  ampHostStatging: 'amp.s.aws.economist.com',
```
Comment (umbobabo) [inline src/server/config/economist.js]: There is a typo.
BTW I think we should split Stage and Prod in a different way:
```
import here the environment detection....let's say isStage method...
const configs =   {
      prod: {
          domain: 'www.economist.com',  <<< No protocol is better
          ampDomain: 'amp.economist.com'
      },
      stage: {
          domain: 'stage.economist.com',  <<< No protocol is better
          ampDomain: 'amp.s.aws.economist.com'
      },
}
const actualConfiguration = isStage() ? config.stage : config.prod
export default actualConfiguration;
```

Comment (wajerrr) [reply src/server/config/economist.js]: good point
done

**Pull request merged**